<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Palle Villesen" />


<title>Changes in mutational patterns over time</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">sars-cov2-gisaid</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="S2_mutations_counts_covs_exploration_visuals.html">Overview period 1</a>
</li>
<li>
  <a href="S2_mutations_counts_covs_exploration_visuals_period2.html">Overview period 2</a>
</li>
<li>
  <a href="S2_mutations_counts_covs_exploration_visuals_period3.html">Overview period 3</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Changes in mutational patterns over time</h1>
<h4 class="author">Palle Villesen</h4>
<h4 class="date">Last update: 2021-07-01 10:04:39</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>sars-cov2-gisaid/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210408code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210408)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210408code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210408)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong9c3698e"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 9c3698e </a>
</p>
</div>
<div id="strongRepositoryversionstrong9c3698e" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 9c3698e. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/1_working_with_the_data.snpeff.v2.gisaid_v2.0.rmd
    Untracked:  analysis/2_human_lineage_mutations.snpeff.v1.rmd
    Untracked:  analysis/hard_coded_params.R
    Untracked:  code/make_debug_miniature_dataset.R
    Untracked:  code/make_genome_positions.R
    Untracked:  data/2021-04-09/
    Untracked:  data/MutCounts_with_covariates.csv
    Untracked:  data/proteins_sarscov2_genome_positions.rds
    Untracked:  how_to_work_together.md
    Untracked:  output/README.html

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Deleted:    analysis/S0_creating_data_bywindows_lineages_OBSOLETE.Rmd
    Modified:   analysis/S0_creating_data_bywindows_lineages_byphase.Rmd
    Modified:   analysis/S1_adding_covariates_multiple_lineage_early_late.Rmd
    Modified:   analysis/S1_adding_covariates_multiple_lineage_period2.Rmd
    Modified:   analysis/S1_adding_covariates_multiple_lineage_period3.Rmd
    Modified:   analysis/S2_mutations_counts_covs_exploration_visuals.Rmd
    Modified:   analysis/S2_mutations_counts_covs_exploration_visuals_period2.Rmd
    Modified:   analysis/S2_mutations_counts_covs_exploration_visuals_period3.Rmd
    Modified:   analysis/S4_mutation_counts_models_period1.Rmd
    Modified:   analysis/S4_mutation_counts_models_period2.Rmd
    Modified:   analysis/S4_mutation_counts_models_period3.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/README.md
    Modified:   data/README.md
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<pre class="r"><code>source(&quot;analysis/hard_coded_params.R&quot;)
# data_path &lt;- &quot;data/2021-06-18/&quot; 
# use_downsampled &lt;- TRUE</code></pre>
<div id="using-data-version-data2021-06-18" class="section level1">
<h1>Using data version: data/2021-06-18/</h1>
</div>
<div id="init" class="section level1">
<h1>Init</h1>
<pre class="r"><code>library(tidyverse)
library(cowplot)  # For theme
library(lubridate)
library(ggsci)
library(ggrepel)


text_base_size   &lt;- 8
fig.witdh        &lt;- 210
fig.height       &lt;- 160

ggplot_text_size &lt;- text_base_size / ggplot2::.pt
# Now use: geom_text(..., size = ggplot_text_size)
theme_set(theme_cowplot(font_size = text_base_size, 
                        rel_small = 1, rel_tiny = 1, 
                        rel_large = 1))

knitr::opts_chunk$set(fig.width = fig.witdh/25.4)
knitr::opts_chunk$set(fig.height = fig.height/25.4)
knitr::opts_chunk$set(dpi = 108)

if (use_downsampled) {
  mutations &lt;- read_rds(paste(data_path, &quot;/mutations_snpeff_annotated_tidy_300K_downsampled.rds&quot;, sep=&quot;&quot;))
  metadata  &lt;- read_rds(paste(data_path, &quot;/metadata_snpeff_tidy_300K_downsampled.rds&quot;,sep=&quot;&quot;))
  } else {
  mutations &lt;- read_rds(paste(data_path, &quot;/mutations_snpeff_annotated_tidy.rds&quot;, sep=&quot;&quot;))
  metadata  &lt;- read_rds(paste(data_path, &quot;/metadata_snpeff_tidy.rds&quot;,sep=&quot;&quot;))
}

x &lt;- Sys.setlocale(locale = &quot;English&quot;)
rm(x)</code></pre>
</div>
<div id="mutations-in-specific-countries" class="section level1">
<h1>Mutations in specific countries</h1>
<pre class="r"><code># Get mutated genomes in spike
interesting_mutations &lt;- c(&quot;D614G&quot;, &quot;N501Y&quot;, &quot;E484K&quot;, &quot;E484Q&quot;, &quot;T478K&quot;,&quot;L452R&quot;, &quot;P681R&quot;) #
interesting_countries &lt;- c(&quot;United Kingdom&quot;, &quot;Denmark&quot;, &quot;Portugal&quot;)

transitions   &lt;- c(&quot;C&gt;T&quot;, &quot;T&gt;C&quot;, &quot;A&gt;G&quot;, &quot;G&gt;A&quot;)
transversions &lt;- c(&quot;C&gt;A&quot;, &quot;C&gt;G&quot;, &quot;T&gt;A&quot;, &quot;T&gt;G&quot;, &quot;A&gt;C&quot;, &quot;A&gt;T&quot;, &quot;G&gt;T&quot;, &quot;G&gt;C&quot;)

interesting_lineages &lt;- c(&quot;B.1&quot;,
                          &quot;B.1.1.7&quot;,# British
                          &quot;B.1.351&quot;, # South Africa
                          &quot;B.1.617.2&quot; # Indian variant
                          )

# Tag each genome with each of these mutations
tmp1 &lt;- metadata %&gt;%
  filter(species==&quot;Human&quot; &amp; country %in% interesting_countries) %&gt;%
  mutate(date_bin = cut(date, breaks = &quot;7 days&quot;) %&gt;% ymd())
  
# For each mutation, get proportion

results &lt;- tibble()

for(i in 1:length(interesting_mutations)) {
  
  tmp2 &lt;- mutations %&gt;%
    filter(gene==&quot;S&quot; &amp; aa_change==interesting_mutations[i])
  
  tmp3 &lt;- tmp1 %&gt;%
    mutate(mutation = interesting_mutations[i]) %&gt;%
    mutate(has_it = ifelse(id %in% tmp2$id, &quot;mutated&quot;, &quot;not_mutated&quot;)) %&gt;%
    group_by(country, date_bin, mutation) %&gt;%
     summarise(total = n(),
               prop = sum(has_it==&quot;mutated&quot;) / n())

  results &lt;- results %&gt;% bind_rows(tmp3)
}</code></pre>
<pre><code>`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)
`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)
`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)
`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)
`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)
`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)
`summarise()` regrouping output by &#39;country&#39;, &#39;date_bin&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code>pd &lt;- results

ggplot(pd, aes(x=date_bin, y=prop,size=total, color=mutation)) + 
  geom_point() +
  geom_line(size=1) +
  xlab(&quot;&quot;) +
  ylab(&quot;Proportion of genomes with mutation&quot;) +
  scale_size(&quot;Number of genomes&quot;) +
  ggsci::scale_color_aaas(name=&quot;&quot;) +
  facet_wrap( ~country, ncol=2) +
  NULL</code></pre>
<pre><code>Warning: Removed 7 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 7 row(s) containing missing values (geom_path).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-3-1.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
</div>
<div id="lineages-in-dk-uk" class="section level1">
<h1>Lineages in DK &amp; UK</h1>
<pre class="r"><code>tmp &lt;- metadata %&gt;% 
  filter(species==&quot;Human&quot; &amp; country %in% interesting_countries) %&gt;%
  mutate(pangolin2 = ifelse(pangolin_lineage %in% interesting_lineages, pangolin_lineage, &quot;Other&quot;)) %&gt;%
  mutate(date_bin = cut(date, breaks = &quot;7 days&quot;) %&gt;% ymd()) %&gt;%
  group_by(date_bin, country) %&gt;%
  count(pangolin2) %&gt;%
  {.}

pd &lt;- tmp %&gt;% 
  group_by(date_bin, country) %&gt;%
  summarise(total = sum(n)) %&gt;%
  left_join(tmp) %&gt;%
  mutate(prop = n/total)</code></pre>
<pre><code>`summarise()` regrouping output by &#39;date_bin&#39; (override with `.groups` argument)</code></pre>
<pre><code>Joining, by = c(&quot;date_bin&quot;, &quot;country&quot;)</code></pre>
<pre class="r"><code>ggplot(pd, aes(x=date_bin, y=prop, size=total, color=pangolin2)) + 
  geom_point() +
  geom_line(size=1) +
  xlab(&quot;&quot;) +
  ylab(&quot;Proportion of genomes with mutation&quot;) +
  scale_size(&quot;Number of genomes&quot;) +
  ggsci::scale_color_aaas(name=&quot;&quot;) +
  facet_wrap(~country, ncol=2) +
  NULL</code></pre>
<pre><code>Warning: Removed 5 rows containing missing values (geom_point).</code></pre>
<pre><code>Warning: Removed 5 row(s) containing missing values (geom_path).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-4-1.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
</div>
<div id="dk-analysis-mutational-clock-syn-vs.-non-syn" class="section level1">
<h1>DK analysis: Mutational clock (Syn vs. Non-Syn)</h1>
<pre class="r"><code>tmp &lt;- metadata %&gt;% 
  filter(species==&quot;Human&quot; &amp; country %in% interesting_countries) %&gt;%
  filter(pangolin_lineage %in% interesting_lineages) %&gt;%
  mutate(date_bin = cut(date, breaks = &quot;2 weeks&quot;) %&gt;% ymd()) %&gt;%
  {.}

pd &lt;-  tmp %&gt;%
  group_by(date_bin,country, pangolin_lineage, count_N, count_S) %&gt;%
  count() %&gt;%
  pivot_longer(names_to = &quot;type&quot;, 
              values_to = &quot;mutations&quot;, 
                   cols = c(count_N, count_S)) %&gt;%
  mutate(type = fct_recode(type, 
                           &quot;Synonymous mutations&quot;=&quot;count_S&quot;,
                           &quot;Nonsynonymous mutations&quot;=&quot;count_N&quot;
                           ))

ggplot(pd, aes(x=date_bin, y=mutations, color=pangolin_lineage, size=n)) +
  geom_point(alpha=0.5) + 
  geom_smooth(method=&quot;lm&quot;, se=F, size=1, show.legend = F) +
  ylab(&quot;Accumulated mutations&quot;) +
  facet_grid(country ~ type) +
  theme(axis.title.x = element_blank()) +
  scale_color_aaas(name=&quot;Lineage&quot;) +
  scale_size_continuous(name = &quot;Genomes&quot;) +
  theme(strip.background = element_blank()) +
  ggtitle(paste(nrow(tmp), &quot; genomes&quot;, sep=&quot;&quot;)) +
  NULL</code></pre>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<pre><code>Warning: Removed 158 rows containing non-finite values (stat_smooth).</code></pre>
<pre><code>Warning: Removed 158 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-5-1.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
<div id="lineage-analysis" class="section level2">
<h2>Lineage analysis</h2>
<p>Remove nearly fixed mutations</p>
<pre class="r"><code>tmp1 &lt;- metadata %&gt;% 
  filter(species==&quot;Human&quot; &amp; country %in% interesting_countries) %&gt;%
  filter(pangolin_lineage %in% interesting_lineages) %&gt;%
  mutate(date_bin = cut(date, breaks = &quot;2 weeks&quot;) %&gt;% ymd())

# Genomes pr lineage
tmp2 &lt;- tmp1 %&gt;%
  group_by(pangolin_lineage) %&gt;%
  summarise(genomes=n())</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code># count of genomes mutated pr mutation
tmp3 &lt;- tmp1 %&gt;% 
  select(id, pangolin_lineage) %&gt;%
  left_join(mutations) %&gt;% 
  select(pangolin_lineage, id, ref_base, variant_base, position) %&gt;%
  distinct() %&gt;%
  group_by(pangolin_lineage, ref_base, variant_base, position) %&gt;%
  summarise(genomes_mutated = n()) %&gt;%
  mutate(base_change = paste(ref_base, variant_base, sep=&quot;&gt;&quot;)) %&gt;%
  mutate(mutation_type = ifelse(base_change %in% transitions, &quot;transition&quot;, NA)) %&gt;%
  mutate(mutation_type = ifelse(base_change %in% transversions, &quot;transversion&quot;, mutation_type))</code></pre>
<pre><code>Joining, by = &quot;id&quot;</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;ref_base&#39;, &#39;variant_base&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code># Good mutations
tmp4 &lt;- tmp2 %&gt;% 
  left_join(tmp3) %&gt;% 
  filter(genomes_mutated/genomes &lt; 0.9)</code></pre>
<pre><code>Joining, by = &quot;pangolin_lineage&quot;</code></pre>
<pre class="r"><code># Getting summary of each genome
tmp5 &lt;- tmp4 %&gt;% 
  left_join(mutations) %&gt;% # This will limit to only the good mutations
  filter(gene!=&quot;ORF1ab_pp1a&quot;) %&gt;%
  group_by(id) %&gt;% # Summarise pr. genome
  summarise(count_ts = sum(base_change %in% transitions),
            count_tv = sum(base_change %in% transversions),
            count_non = sum(type==&quot;N&quot;),
            count_syn = sum(type==&quot;S&quot;)
            ) %&gt;%
  filter(count_syn &gt; 5) %&gt;%
  filter(count_tv &gt; 5) %&gt;%
  mutate(tstv_ratio = count_ts/count_tv) %&gt;%
  mutate(ns_ratio = count_non/count_syn) %&gt;%
  filter(tstv_ratio &gt; 0) %&gt;%
  filter(ns_ratio &gt; 0) %&gt;%
  {.}</code></pre>
<pre><code>Joining, by = c(&quot;ref_base&quot;, &quot;variant_base&quot;, &quot;position&quot;)</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<pre class="r"><code>pd &lt;- tmp1 %&gt;% left_join(tmp5)</code></pre>
<pre><code>Joining, by = &quot;id&quot;</code></pre>
<pre class="r"><code>ggplot(pd,aes(x=date_bin, y=tstv_ratio)) + 
  geom_jitter(width=1, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), 
               fun=median, colour=&quot;red&quot;, geom=&quot;line&quot;) +
  facet_grid(country ~ pangolin_lineage) +
  ylab(&quot;Nonsynonymous / synonymous changes&quot;) +
  theme(axis.title.x = element_blank())+
  NULL</code></pre>
<pre><code>Warning: Removed 11875 rows containing non-finite values (stat_summary).</code></pre>
<pre><code>Warning: Removed 11875 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-6-1.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(pd,aes(x=date_bin, y=ns_ratio)) + 
  geom_jitter(width=1, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), fun=median, colour=&quot;red&quot;, geom=&quot;line&quot;) +
  facet_grid(country ~ pangolin_lineage) +
  ylab(&quot;Transitions / Transversions&quot;) +
  theme(axis.title.x = element_blank()) +
  NULL</code></pre>
<pre><code>Warning: Removed 11875 rows containing non-finite values (stat_summary).

Warning: Removed 11875 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-6-2.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pd2 &lt;- pd %&gt;% pivot_longer(cols = c(count_ts, count_tv))

ggplot(pd2,aes(x=date_bin, y=value)) + 
  geom_jitter(width=0.5, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), fun=median, colour=&quot;red&quot;, geom=&quot;line&quot;) +
  facet_grid(name~pangolin_lineage) +
  theme(axis.title = element_blank()) +
  NULL</code></pre>
<pre><code>Warning: Removed 23750 rows containing non-finite values (stat_summary).</code></pre>
<pre><code>Warning: Removed 23750 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-6-3.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pd2 &lt;- pd %&gt;% pivot_longer(cols = c(count_syn, count_non))

ggplot(pd2,aes(x=date_bin, y=value)) + 
  geom_jitter(width=1, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), fun=median, colour=&quot;red&quot;, geom=&quot;line&quot;) +
  facet_grid(name~pangolin_lineage) +
  theme(axis.title = element_blank()) +
  NULL</code></pre>
<pre><code>Warning: Removed 23750 rows containing non-finite values (stat_summary).

Warning: Removed 23750 rows containing missing values (geom_point).</code></pre>
<p><img src="figure/2_human_lineage_mutations.snpeff.v1.rmd/unnamed-chunk-6-4.png" width="892.913385826772" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.1   ggsci_2.9       lubridate_1.7.9 cowplot_1.1.0  
 [5] forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2     purrr_0.3.4    
 [9] readr_1.3.1     tidyr_1.1.1     tibble_3.0.3    ggplot2_3.3.2  
[13] tidyverse_1.3.0 workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       lattice_0.20-41  assertthat_0.2.1 rprojroot_2.0.2 
 [5] digest_0.6.25    R6_2.4.1         cellranger_1.1.0 backports_1.1.9 
 [9] reprex_0.3.0     evaluate_0.14    httr_1.4.2       pillar_1.4.6    
[13] rlang_0.4.7      readxl_1.3.1     rstudioapi_0.11  blob_1.2.1      
[17] Matrix_1.2-18    rmarkdown_2.3    labeling_0.3     splines_4.0.2   
[21] munsell_0.5.0    broom_0.7.0      compiler_4.0.2   httpuv_1.5.4    
[25] modelr_0.1.8     xfun_0.16        pkgconfig_2.0.3  mgcv_1.8-31     
[29] htmltools_0.5.0  tidyselect_1.1.0 fansi_0.4.1      crayon_1.3.4    
[33] dbplyr_1.4.4     withr_2.2.0      later_1.1.0.1    grid_4.0.2      
[37] nlme_3.1-148     jsonlite_1.7.1   gtable_0.3.0     lifecycle_0.2.0 
[41] DBI_1.1.0        git2r_0.27.1     magrittr_1.5     scales_1.1.1    
[45] cli_2.0.2        stringi_1.4.6    farver_2.0.3     fs_1.5.0        
[49] promises_1.1.1   xml2_1.3.2       ellipsis_0.3.1   generics_0.0.2  
[53] vctrs_0.3.2      tools_4.0.2      glue_1.4.1       hms_0.5.3       
[57] yaml_2.2.1       colorspace_1.4-1 rvest_0.3.6      knitr_1.29      
[61] haven_2.3.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
