<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Thomas Bataillon reusing PV examples" />


<title>Extracting windows based counts of mutations from gisaid data freeze parsed by the snpEff pipeline</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">sars-cov2-gisaid</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="S2_mutations_counts_covs_exploration_visuals.html">Overview period 1</a>
</li>
<li>
  <a href="S2_mutations_counts_covs_exploration_visuals_period2.html">Overview period 2</a>
</li>
<li>
  <a href="S2_mutations_counts_covs_exploration_visuals_period3.html">Overview period 3</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Extracting windows based counts of mutations from gisaid data freeze parsed by the snpEff pipeline</h1>
<h4 class="author">Thomas Bataillon reusing PV examples</h4>
<h4 class="date">Last update: 2021-06-30 13:46:43</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>sars-cov2-gisaid/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210408code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210408)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210408code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210408)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong9c3698e"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 9c3698e </a>
</p>
</div>
<div id="strongRepositoryversionstrong9c3698e" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 9c3698e. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/1_working_with_the_data.snpeff.v2.gisaid_v2.0.rmd
    Untracked:  analysis/2_human_lineage_mutations.snpeff.v1.rmd
    Untracked:  analysis/hard_coded_params.R
    Untracked:  code/make_debug_miniature_dataset.R
    Untracked:  code/make_genome_positions.R
    Untracked:  data/2021-04-09/
    Untracked:  data/MutCounts_with_covariates.csv
    Untracked:  data/proteins_sarscov2_genome_positions.rds
    Untracked:  how_to_work_together.md
    Untracked:  output/README.html

Unstaged changes:
    Modified:   README.md
    Modified:   _workflowr.yml
    Deleted:    analysis/S0_creating_data_bywindows_lineages_OBSOLETE.Rmd
    Modified:   analysis/S0_creating_data_bywindows_lineages_byphase.Rmd
    Modified:   analysis/S2_mutations_counts_covs_exploration_visuals.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/README.md
    Modified:   data/README.md
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/S0_creating_data_bywindows_lineages_byphase.Rmd</code>) and HTML (<code>docs/S0_creating_data_bywindows_lineages_byphase.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
9c3698e
</td>
<td>
tbata
</td>
<td>
2021-06-30
</td>
<td>
Added 100 nt windows for 3 periods on downsized 300k data. and update of all analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overview" class="section level1">
<h1>Overview</h1>
<ul>
<li>We filter away a few genomes (wrong date etc).</li>
<li>We create summaries binning by genomic windows starting at each gene, stratifying by pangolin lineages</li>
<li>we recode the gene name according to their genomic positions</li>
<li>We output counts as rds files for each phase (time peridod) separately</li>
</ul>
<pre class="r"><code># data_path &lt;- &quot;data/2021-06-18/&quot; # Now obsolete as hard coded

source(&quot;analysis/hard_coded_params.R&quot;)</code></pre>
</div>
<div id="using-data-version-data2021-06-18" class="section level1">
<h1>Using data version: data/2021-06-18/</h1>
</div>
<div id="init-loading-libraries-and-data" class="section level1">
<h1>Init: loading libraries and data</h1>
<pre class="r"><code>library(tidyverse)
library(cowplot)  # For theme
library(lubridate)
library(ggsci)
library(ggrepel)

text_base_size   &lt;- 10   # in pt
ggplot_text_size &lt;- text_base_size / ggplot2::.pt
# Now use: geom_text(..., size = ggplot_text_size)
theme_set(theme_cowplot(font_size = text_base_size, 
                        rel_small = 1, rel_tiny = 1, 
                        rel_large = 1))


# mutations    &lt;- read_rds(paste(data_path, &quot;/mutations_snpeff_annotated_tidy.rds&quot;, sep=&quot;&quot;))

#just for debug .. to be updated once coding works  TB May 19, 2021

# mutations &lt;- read_rds(paste(data_path, &quot;/mutations_miniature_snpeff_annotated_tidy.rds&quot;, sep=&quot;&quot;))
# dim(mutations)
# metadata  &lt;- read_rds(paste(data_path, &quot;/metadata_snpeff_tidy.rds&quot;,sep=&quot;&quot;)) 
# 

tt1 &lt;- Sys.time()
mutations    &lt;- read_rds(paste(data_path, &quot;/mutations_snpeff_annotated_tidy_300K_downsampled.rds&quot;, sep=&quot;&quot;))
Sys.time() - tt1</code></pre>
<pre><code>Time difference of 33.83882 secs</code></pre>
<pre class="r"><code>metadata  &lt;- read_rds(paste(data_path, &quot;/metadata_snpeff_tidy_300K_downsampled.rds&quot;,sep=&quot;&quot;)) 

dim(metadata)</code></pre>
<pre><code>[1] 300000     37</code></pre>
<pre class="r"><code>genomes_md &lt;- metadata %&gt;% 
  # left_join(phylogeny_md) %&gt;% #now obsolete as we have a single metadata rds
  mutate(date = ymd(date)) %&gt;%
  filter(species == &quot;Human&quot;) %&gt;%
  mutate(date_sub = str_sub(date,1,4)) %&gt;%
  filter(date_sub  %in% c(&quot;2019&quot;,&quot;2020&quot;,&quot;2021&quot;)) %&gt;%
  {.}

rm(metadata) #cleaning up</code></pre>
</div>
<div id="a-quick-view-of-the-merged" class="section level1">
<h1>A quick view of the merged</h1>
</div>
<div id="selecting-a-set-of-lineages" class="section level1">
<h1>Selecting a set of lineages</h1>
<p>Note this new version: referencing according to Susan data integration We select a subset of lineages <code>to_extract</code>: These are offering a view of the worldwide pandemic along the different historical time periods For the remaining of the analysis we only use thes specfific lineages Note I only found 1 variant matching “B.1.617” (to be checked)</p>
<pre class="r"><code>to_extract&lt;- pangolin_lineages

  # c(&quot;B.1.177&quot;, &quot;B.1.5&quot;, &quot;B.1&quot;, &quot;B.1.1&quot;,&quot;B.1.1.28&quot;,  &quot;B.1.1.7&quot;, &quot;B.1.351&quot;, &quot;P.1&quot;)

genomes_md&lt;-genomes_md %&gt;%
  filter(pangolin_lineage %in% to_extract) 

dim(genomes_md)</code></pre>
<pre><code>[1] 173991     38</code></pre>
<pre class="r"><code>genomes_md %&gt;%
    group_by(pangolin_lineage) %&gt;%
  summarise(n= n(),
            birth = min(`Collection date`)) %&gt;%
  knitr::kable()</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">pangolin_lineage</th>
<th align="right">n</th>
<th align="left">birth</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B.1</td>
<td align="right">10196</td>
<td align="left">2020-02-01</td>
</tr>
<tr class="even">
<td align="left">B.1.1</td>
<td align="right">6318</td>
<td align="left">2020-02-24</td>
</tr>
<tr class="odd">
<td align="left">B.1.1.28</td>
<td align="right">362</td>
<td align="left">2020-03-09</td>
</tr>
<tr class="even">
<td align="left">B.1.1.7</td>
<td align="right">138798</td>
<td align="left">2020-04-26</td>
</tr>
<tr class="odd">
<td align="left">B.1.177</td>
<td align="right">10778</td>
<td align="left">2020-03-24</td>
</tr>
<tr class="even">
<td align="left">B.1.351</td>
<td align="right">2809</td>
<td align="left">2020-09-15</td>
</tr>
<tr class="odd">
<td align="left">P.1</td>
<td align="right">4730</td>
<td align="left">2020-10-13</td>
</tr>
</tbody>
</table>
</div>
<div id="adding-genomic-coordinates-that-are-gene-centric" class="section level1">
<h1>Adding genomic coordinates that are gene-centric</h1>
<p>(Note, here or in the <code>/data/Readme.md</code>, provide details for the data source of <code>output/proteins_sarscov2_genome_positions.rds</code> )</p>
<pre class="r"><code>gene_positions_md &lt;-readRDS(&quot;data/proteins_sarscov2_genome_positions.rds&quot;)
gene_positions_md %&gt;% arrange(start_pos) %&gt;%knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">name</th>
<th align="right">start_pos</th>
<th align="right">end_pos</th>
<th align="right">gene_length</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">leader</td>
<td align="right">266</td>
<td align="right">806</td>
<td align="right">540</td>
</tr>
<tr class="even">
<td align="left">nsp2</td>
<td align="right">806</td>
<td align="right">2720</td>
<td align="right">1914</td>
</tr>
<tr class="odd">
<td align="left">nsp3</td>
<td align="right">2720</td>
<td align="right">8555</td>
<td align="right">5835</td>
</tr>
<tr class="even">
<td align="left">nsp4</td>
<td align="right">8555</td>
<td align="right">10055</td>
<td align="right">1500</td>
</tr>
<tr class="odd">
<td align="left">proteinase</td>
<td align="right">10055</td>
<td align="right">10973</td>
<td align="right">918</td>
</tr>
<tr class="even">
<td align="left">nsp6</td>
<td align="right">10973</td>
<td align="right">11843</td>
<td align="right">870</td>
</tr>
<tr class="odd">
<td align="left">nsp7</td>
<td align="right">11843</td>
<td align="right">12092</td>
<td align="right">249</td>
</tr>
<tr class="even">
<td align="left">nsp8</td>
<td align="right">12092</td>
<td align="right">12686</td>
<td align="right">594</td>
</tr>
<tr class="odd">
<td align="left">nsp9</td>
<td align="right">12686</td>
<td align="right">13025</td>
<td align="right">339</td>
</tr>
<tr class="even">
<td align="left">nsp10</td>
<td align="right">13025</td>
<td align="right">13442</td>
<td align="right">417</td>
</tr>
<tr class="odd">
<td align="left">nsp11</td>
<td align="right">13442</td>
<td align="right">13481</td>
<td align="right">39</td>
</tr>
<tr class="even">
<td align="left">RDRp</td>
<td align="right">13442</td>
<td align="right">16238</td>
<td align="right">2796</td>
</tr>
<tr class="odd">
<td align="left">helicase</td>
<td align="right">16237</td>
<td align="right">18040</td>
<td align="right">1803</td>
</tr>
<tr class="even">
<td align="left">exonuclease</td>
<td align="right">18040</td>
<td align="right">19621</td>
<td align="right">1581</td>
</tr>
<tr class="odd">
<td align="left">endoRNAse</td>
<td align="right">19621</td>
<td align="right">20659</td>
<td align="right">1038</td>
</tr>
<tr class="even">
<td align="left">methyltransferase</td>
<td align="right">20659</td>
<td align="right">21550</td>
<td align="right">891</td>
</tr>
<tr class="odd">
<td align="left">S</td>
<td align="right">21563</td>
<td align="right">25382</td>
<td align="right">3819</td>
</tr>
<tr class="even">
<td align="left">orf3a</td>
<td align="right">25393</td>
<td align="right">26218</td>
<td align="right">825</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="right">26245</td>
<td align="right">26470</td>
<td align="right">225</td>
</tr>
<tr class="even">
<td align="left">M</td>
<td align="right">26523</td>
<td align="right">27189</td>
<td align="right">666</td>
</tr>
<tr class="odd">
<td align="left">orf6</td>
<td align="right">27202</td>
<td align="right">27385</td>
<td align="right">183</td>
</tr>
<tr class="even">
<td align="left">orf7a</td>
<td align="right">27394</td>
<td align="right">27757</td>
<td align="right">363</td>
</tr>
<tr class="odd">
<td align="left">orf7b</td>
<td align="right">27756</td>
<td align="right">27885</td>
<td align="right">129</td>
</tr>
<tr class="even">
<td align="left">orf8</td>
<td align="right">27894</td>
<td align="right">28257</td>
<td align="right">363</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="right">28274</td>
<td align="right">29531</td>
<td align="right">1257</td>
</tr>
<tr class="even">
<td align="left">orf10</td>
<td align="right">29558</td>
<td align="right">29672</td>
<td align="right">114</td>
</tr>
</tbody>
</table>
<p>Split into 100 bp windows in genome I position bins according to the gene starts to get a <code>position_bin</code> that in principle should match the ones computed by Susan in the co-variate files : <code>genome_bin</code> gives genomic windows numbered consecutively 1, 2, 3… <code>window_start</code> provides the genomic coordinate of each <code>genome_bin</code> Finally <code>gene_bin</code> renames consecutively the windows 1,2, 3, … within each gene.</p>
<p>Having both coordinate system allows - in principle - to retain some level of sanity while juggling with the co-variates (see following script S1 for aggregating all co-variates compiled during the summer):</p>
<pre class="r"><code>names(mutations)</code></pre>
<pre><code> [1] &quot;id&quot;           &quot;CHROM&quot;        &quot;position&quot;     &quot;ref_base&quot;     &quot;variant_base&quot;
 [6] &quot;gene&quot;         &quot;biotype&quot;      &quot;impact&quot;       &quot;effect&quot;       &quot;hgvs_c&quot;      
[11] &quot;hgvs_p&quot;       &quot;aa_position&quot;  &quot;type&quot;         &quot;aa_change&quot;   </code></pre>
<pre class="r"><code>unique(mutations$gene)</code></pre>
<pre><code> [1] &quot;ORF1ab_pp1a&quot;  &quot;ORF1ab_pp1ab&quot; &quot;S&quot;            &quot;M&quot;            &quot;ORF8&quot;        
 [6] &quot;N&quot;            &quot;ORF10&quot;        &quot;ORF3a&quot;        &quot;ORF7a&quot;        &quot;E&quot;           
[11] &quot;ORF6&quot;         &quot;ORF7b&quot;       </code></pre>
<pre class="r"><code>mutations$gene &lt;- tolower(mutations$gene)
unique(mutations$gene)</code></pre>
<pre><code> [1] &quot;orf1ab_pp1a&quot;  &quot;orf1ab_pp1ab&quot; &quot;s&quot;            &quot;m&quot;            &quot;orf8&quot;        
 [6] &quot;n&quot;            &quot;orf10&quot;        &quot;orf3a&quot;        &quot;orf7a&quot;        &quot;e&quot;           
[11] &quot;orf6&quot;         &quot;orf7b&quot;       </code></pre>
<pre class="r"><code>names(genomes_md)</code></pre>
<pre><code> [1] &quot;id&quot;                              &quot;Virus name&quot;                     
 [3] &quot;Type&quot;                            &quot;Accession ID&quot;                   
 [5] &quot;Collection date&quot;                 &quot;Location&quot;                       
 [7] &quot;region&quot;                          &quot;country&quot;                        
 [9] &quot;division&quot;                        &quot;Additional location information&quot;
[11] &quot;Sequence length&quot;                 &quot;Host&quot;                           
[13] &quot;Patient age&quot;                     &quot;Gender&quot;                         
[15] &quot;Clade&quot;                           &quot;pangolin_lineage&quot;               
[17] &quot;Pangolin version&quot;                &quot;Variant&quot;                        
[19] &quot;AA Substitutions&quot;                &quot;date_submitted&quot;                 
[21] &quot;Is reference?&quot;                   &quot;Is complete?&quot;                   
[23] &quot;Is high coverage?&quot;               &quot;Is low coverage?&quot;               
[25] &quot;N-Content&quot;                       &quot;GC-Content&quot;                     
[27] &quot;species&quot;                         &quot;date&quot;                           
[29] &quot;Patient status&quot;                  &quot;Passage&quot;                        
[31] &quot;Specimen&quot;                        &quot;Additional host information&quot;    
[33] &quot;Lineage&quot;                         &quot;Sampling strategy&quot;              
[35] &quot;Last vaccinated&quot;                 &quot;count_N&quot;                        
[37] &quot;count_S&quot;                         &quot;date_sub&quot;                       </code></pre>
<pre class="r"><code>names(gene_positions_md)[1]&lt;-&quot;gene&quot; # Renaming the gene column to make it compatible with the column name in the mutation tibble

unique(mutations$gene)</code></pre>
<pre><code> [1] &quot;orf1ab_pp1a&quot;  &quot;orf1ab_pp1ab&quot; &quot;s&quot;            &quot;m&quot;            &quot;orf8&quot;        
 [6] &quot;n&quot;            &quot;orf10&quot;        &quot;orf3a&quot;        &quot;orf7a&quot;        &quot;e&quot;           
[11] &quot;orf6&quot;         &quot;orf7b&quot;       </code></pre>
<pre class="r"><code>unique(gene_positions_md$gene)</code></pre>
<pre><code> [1] &quot;E&quot;                 &quot;endoRNAse&quot;         &quot;exonuclease&quot;      
 [4] &quot;helicase&quot;          &quot;leader&quot;            &quot;M&quot;                
 [7] &quot;methyltransferase&quot; &quot;N&quot;                 &quot;nsp10&quot;            
[10] &quot;nsp11&quot;             &quot;nsp2&quot;              &quot;nsp3&quot;             
[13] &quot;nsp4&quot;              &quot;nsp6&quot;              &quot;nsp7&quot;             
[16] &quot;nsp8&quot;              &quot;nsp9&quot;              &quot;orf10&quot;            
[19] &quot;orf3a&quot;             &quot;orf6&quot;              &quot;orf7a&quot;            
[22] &quot;orf7b&quot;             &quot;orf8&quot;              &quot;proteinase&quot;       
[25] &quot;RDRp&quot;              &quot;S&quot;                </code></pre>
<div id="adding-extra-gene-names" class="section level2">
<h2>adding extra gene names</h2>
<p>We use an ad hoc recoding to associate alt gene name nomenclature and check that the number of mutations is conserved across both nomencaltures</p>
<pre class="r"><code>source(&quot;code/make_genome_positions.R&quot;)
get_gene_name_pv_tb(13490) #check RDp</code></pre>
<pre><code>[1] &quot;RDRp&quot;</code></pre>
<pre class="r"><code>get_gene_name_pv_tb(28000) #check orf8</code></pre>
<pre><code>[1] &quot;orf8&quot;</code></pre>
<pre class="r"><code>initial_time = Sys.time()
mutations$alt_gene_names &lt;- sapply(X = mutations$position, get_gene_name_pv_tb)
final_time = Sys.time()

final_time - initial_time</code></pre>
<pre><code>Time difference of 1.890087 mins</code></pre>
<pre class="r"><code>table(mutations$alt_gene_names)</code></pre>
<pre><code>
                E         endoRNAse       exonuclease          helicase 
            14550             77099            136060            180766 
           leader                 M methyltransferase                 N 
           146452            106734            101479           1509144 
    No gene found             nsp10             nsp11              nsp2 
              245             35628              3644            763740 
             nsp3              nsp4              nsp6              nsp7 
          2813350            283778            198980             37772 
             nsp8              nsp9             orf10             orf3a 
            84706             85024             35806            237763 
             orf6             orf7a             orf7b              orf8 
            15568             58638             20839            404082 
       proteinase              RDRp                 S 
           173288            954835           1749871 </code></pre>
<pre class="r"><code>table(mutations$gene)</code></pre>
<pre><code>
           e            m            n        orf10  orf1ab_pp1a orf1ab_pp1ab 
       14550       106734      1509144        35806      2313185      3763614 
       orf3a         orf6        orf7a        orf7b         orf8            s 
      237763        15568        58685        20839       404082      1749871 </code></pre>
<pre class="r"><code>sum(table(mutations$alt_gene_names))</code></pre>
<pre><code>[1] 10229841</code></pre>
<pre class="r"><code>sum(table(mutations$gene))</code></pre>
<pre><code>[1] 10229841</code></pre>
<pre class="r"><code>names(mutations)</code></pre>
<pre><code> [1] &quot;id&quot;             &quot;CHROM&quot;          &quot;position&quot;       &quot;ref_base&quot;      
 [5] &quot;variant_base&quot;   &quot;gene&quot;           &quot;biotype&quot;        &quot;impact&quot;        
 [9] &quot;effect&quot;         &quot;hgvs_c&quot;         &quot;hgvs_p&quot;         &quot;aa_position&quot;   
[13] &quot;type&quot;           &quot;aa_change&quot;      &quot;alt_gene_names&quot;</code></pre>
<pre class="r"><code>head(mutations)</code></pre>
<pre><code># A tibble: 6 x 15
  id    CHROM position ref_base variant_base gene  biotype impact effect hgvs_c
  &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt; 
1 1000… NC_0…      445 T        C            orf1… protei… LOW    synon… c.180…
2 1000… NC_0…      445 T        C            orf1… protei… LOW    synon… c.180…
3 1000… NC_0…     3037 C        T            orf1… protei… LOW    synon… c.277…
4 1000… NC_0…     3037 C        T            orf1… protei… LOW    synon… c.277…
5 1000… NC_0…     6286 C        T            orf1… protei… LOW    synon… c.602…
6 1000… NC_0…     6286 C        T            orf1… protei… LOW    synon… c.602…
# … with 5 more variables: hgvs_p &lt;chr&gt;, aa_position &lt;chr&gt;, type &lt;chr&gt;,
#   aa_change &lt;chr&gt;, alt_gene_names &lt;chr&gt;</code></pre>
<pre class="r"><code>mutations$gene &lt;- mutations$alt_gene_names ## Ensuring a smooth left_join below</code></pre>
<pre class="r"><code>tt&lt;-Sys.time()
mutations_annotated &lt;- mutations %&gt;% 
  left_join(genomes_md) %&gt;%
  left_join(gene_positions_md) %&gt;%
  mutate(genome_bin = position %/% 100) %&gt;% # abs genomic numbering 1, 2, etc.
  mutate(window_start = genome_bin*100) %&gt;% # genomic coordinate
  mutate(gene_bin = 1+ (position - start_pos) %/%100 ) %&gt;% #new calculation to get the bin number within each gene
  filter(!is.na(species)) %&gt;%
  {.}</code></pre>
<pre><code>Joining, by = &quot;id&quot;</code></pre>
<pre><code>Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code>Sys.time() -tt</code></pre>
<pre><code>Time difference of 20.17639 secs</code></pre>
<pre class="r"><code>sum(table(mutations_annotated$gene))</code></pre>
<pre><code>[1] 6808767</code></pre>
<pre class="r"><code>sum(table(mutations_annotated$alt_gene_names))</code></pre>
<pre><code>[1] 6808767</code></pre>
<pre class="r"><code>rm(mutations)</code></pre>
<p>A quick visual check that left join worked</p>
<pre class="r"><code>dim(mutations_annotated)</code></pre>
<pre><code>[1] 6808767      58</code></pre>
<pre class="r"><code>names(mutations_annotated)</code></pre>
<pre><code> [1] &quot;id&quot;                              &quot;CHROM&quot;                          
 [3] &quot;position&quot;                        &quot;ref_base&quot;                       
 [5] &quot;variant_base&quot;                    &quot;gene&quot;                           
 [7] &quot;biotype&quot;                         &quot;impact&quot;                         
 [9] &quot;effect&quot;                          &quot;hgvs_c&quot;                         
[11] &quot;hgvs_p&quot;                          &quot;aa_position&quot;                    
[13] &quot;type&quot;                            &quot;aa_change&quot;                      
[15] &quot;alt_gene_names&quot;                  &quot;Virus name&quot;                     
[17] &quot;Type&quot;                            &quot;Accession ID&quot;                   
[19] &quot;Collection date&quot;                 &quot;Location&quot;                       
[21] &quot;region&quot;                          &quot;country&quot;                        
[23] &quot;division&quot;                        &quot;Additional location information&quot;
[25] &quot;Sequence length&quot;                 &quot;Host&quot;                           
[27] &quot;Patient age&quot;                     &quot;Gender&quot;                         
[29] &quot;Clade&quot;                           &quot;pangolin_lineage&quot;               
[31] &quot;Pangolin version&quot;                &quot;Variant&quot;                        
[33] &quot;AA Substitutions&quot;                &quot;date_submitted&quot;                 
[35] &quot;Is reference?&quot;                   &quot;Is complete?&quot;                   
[37] &quot;Is high coverage?&quot;               &quot;Is low coverage?&quot;               
[39] &quot;N-Content&quot;                       &quot;GC-Content&quot;                     
[41] &quot;species&quot;                         &quot;date&quot;                           
[43] &quot;Patient status&quot;                  &quot;Passage&quot;                        
[45] &quot;Specimen&quot;                        &quot;Additional host information&quot;    
[47] &quot;Lineage&quot;                         &quot;Sampling strategy&quot;              
[49] &quot;Last vaccinated&quot;                 &quot;count_N&quot;                        
[51] &quot;count_S&quot;                         &quot;date_sub&quot;                       
[53] &quot;start_pos&quot;                       &quot;end_pos&quot;                        
[55] &quot;gene_length&quot;                     &quot;genome_bin&quot;                     
[57] &quot;window_start&quot;                    &quot;gene_bin&quot;                       </code></pre>
<pre class="r"><code>#check that counts by gene name ae identical 
table(mutations_annotated$alt_gene_names)</code></pre>
<pre><code>
                E         endoRNAse       exonuclease          helicase 
             8641             25185             48344             84357 
           leader                 M methyltransferase                 N 
            63314             33891             37657           1163826 
    No gene found             nsp10             nsp11              nsp2 
               80             19478              1232            508610 
             nsp3              nsp4              nsp6              nsp7 
          2011256             89594             75722             16518 
             nsp8              nsp9             orf10             orf3a 
            43742             39194             15217             79238 
             orf6             orf7a             orf7b              orf8 
             7836             21694              5922            335325 
       proteinase              RDRp                 S 
            65596            727697           1279601 </code></pre>
<pre class="r"><code>table(mutations_annotated$gene)</code></pre>
<pre><code>
                E         endoRNAse       exonuclease          helicase 
             8641             25185             48344             84357 
           leader                 M methyltransferase                 N 
            63314             33891             37657           1163826 
    No gene found             nsp10             nsp11              nsp2 
               80             19478              1232            508610 
             nsp3              nsp4              nsp6              nsp7 
          2011256             89594             75722             16518 
             nsp8              nsp9             orf10             orf3a 
            43742             39194             15217             79238 
             orf6             orf7a             orf7b              orf8 
             7836             21694              5922            335325 
       proteinase              RDRp                 S 
            65596            727697           1279601 </code></pre>
<pre class="r"><code># brief glimpse

mutations_annotated[, c(1:6,50:58)]</code></pre>
<pre><code># A tibble: 6,808,767 x 15
   id    CHROM position ref_base variant_base gene  count_N count_S date_sub
   &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;   
 1 1000… NC_0…      913 C        T            nsp2       20      10 2021    
 2 1000… NC_0…      913 C        T            nsp2       20      10 2021    
 3 1000… NC_0…     1223 C        T            nsp2       20      10 2021    
 4 1000… NC_0…     1223 C        T            nsp2       20      10 2021    
 5 1000… NC_0…     3037 C        T            nsp3       20      10 2021    
 6 1000… NC_0…     3037 C        T            nsp3       20      10 2021    
 7 1000… NC_0…     3267 C        T            nsp3       20      10 2021    
 8 1000… NC_0…     3267 C        T            nsp3       20      10 2021    
 9 1000… NC_0…     5388 C        A            nsp3       20      10 2021    
10 1000… NC_0…     5388 C        A            nsp3       20      10 2021    
# … with 6,808,757 more rows, and 6 more variables: start_pos &lt;dbl&gt;,
#   end_pos &lt;dbl&gt;, gene_length &lt;dbl&gt;, genome_bin &lt;dbl&gt;, window_start &lt;dbl&gt;,
#   gene_bin &lt;dbl&gt;</code></pre>
<pre class="r"><code>sum(is.na(mutations_annotated$gene_bin))</code></pre>
<pre><code>[1] 80</code></pre>
<pre class="r"><code>dim(mutations_annotated)</code></pre>
<pre><code>[1] 6808767      58</code></pre>
<pre class="r"><code># Weeding out a few mutations that have non sensical gene bin positions  
mutations_annotated &lt;- mutations_annotated %&gt;%
  filter(gene_bin&gt;0) </code></pre>
</div>
</div>
<div id="overview-of-data-optional" class="section level1">
<h1>Overview of data (optional)</h1>
<pre class="r"><code>pd &lt;- tibble() %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Genomes total&quot;, 
                   &quot;n&quot; = genomes_md %&gt;% select(id) %&gt;% distinct() %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Genome with mutations in protein coding regions&quot;,
                   &quot;n&quot; = mutations_annotated %&gt;% select(id) %&gt;% distinct() %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Unique mutations total&quot;,
                   &quot;n&quot; = mutations_annotated %&gt;% select(position, variant_base) %&gt;% distinct() %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Synonymous mutations&quot;,
                   &quot;n&quot; = mutations_annotated %&gt;% filter(type==&quot;S&quot;) %&gt;% select(position, variant_base) %&gt;% distinct() %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Nonsynonymous mutations&quot;,
                   &quot;n&quot; = mutations_annotated %&gt;% filter(type==&quot;N&quot;) %&gt;% select(position, variant_base) %&gt;% distinct() %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Human genomes&quot;,
                   &quot;n&quot; = genomes_md %&gt;% filter(species==&quot;Human&quot;) %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Mink genomes&quot;,
                   &quot;n&quot; = genomes_md %&gt;% filter(species==&quot;Mink&quot;) %&gt;% count() %&gt;% pull(n) )) %&gt;%
  bind_rows(tibble(&quot;Description&quot; = &quot;Other genomes&quot;,
                   &quot;n&quot; = genomes_md %&gt;% filter(!species %in% c(&quot;Human&quot;,&quot;Mink&quot;)) %&gt;% count() %&gt;% pull(n) )) %&gt;%
  
  bind_rows(tibble(&quot;Description&quot; = &quot;Samples with Patient status&quot;,
                   &quot;n&quot; = genomes_md %&gt;% filter(!is.na(`Patient status`)) %&gt;% count() %&gt;% pull(n) )) %&gt;%
  {.}

knitr::kable(pd)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Description</th>
<th align="right">n</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Genomes total</td>
<td align="right">173991</td>
</tr>
<tr class="even">
<td align="left">Genome with mutations in protein coding regions</td>
<td align="right">161438</td>
</tr>
<tr class="odd">
<td align="left">Unique mutations total</td>
<td align="right">33036</td>
</tr>
<tr class="even">
<td align="left">Synonymous mutations</td>
<td align="right">12201</td>
</tr>
<tr class="odd">
<td align="left">Nonsynonymous mutations</td>
<td align="right">20839</td>
</tr>
<tr class="even">
<td align="left">Human genomes</td>
<td align="right">173991</td>
</tr>
<tr class="odd">
<td align="left">Mink genomes</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Other genomes</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Samples with Patient status</td>
<td align="right">1535</td>
</tr>
</tbody>
</table>
<div id="unique-mutations-per.-pangolin-lineage" class="section level2">
<h2>Unique mutations per. pangolin lineage</h2>
<pre class="r"><code>mutations_annotated %&gt;%
  #group_by(name,type, position) %&gt;% # Unique positions
  group_by(pangolin_lineage,type, position, ref_base, variant_base) %&gt;% # Unique mutations
  summarise(genomes_mutated=n()) %&gt;%
  ungroup() %&gt;%
  group_by(pangolin_lineage) %&gt;%
  summarise(total = sum(genomes_mutated),
                N = sum(type==&quot;N&quot;),
                S = sum(type==&quot;S&quot;),
               NS = N/S) %&gt;%
  arrange(desc(total)) %&gt;%
  slice(1:25) %&gt;%  # keep only the top of the distribution
  knitr::kable()</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;type&#39;, &#39;position&#39;, &#39;ref_base&#39; (override with `.groups` argument)</code></pre>
<pre><code>`summarise()` ungrouping output (override with `.groups` argument)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">pangolin_lineage</th>
<th align="right">total</th>
<th align="right">N</th>
<th align="right">S</th>
<th align="right">NS</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B.1.1.7</td>
<td align="right">6085757</td>
<td align="right">17885</td>
<td align="right">10986</td>
<td align="right">1.627981</td>
</tr>
<tr class="even">
<td align="left">B.1.177</td>
<td align="right">247363</td>
<td align="right">5014</td>
<td align="right">3293</td>
<td align="right">1.522624</td>
</tr>
<tr class="odd">
<td align="left">P.1</td>
<td align="right">219019</td>
<td align="right">3161</td>
<td align="right">2170</td>
<td align="right">1.456682</td>
</tr>
<tr class="even">
<td align="left">B.1.351</td>
<td align="right">92138</td>
<td align="right">2026</td>
<td align="right">1474</td>
<td align="right">1.374491</td>
</tr>
<tr class="odd">
<td align="left">B.1</td>
<td align="right">88341</td>
<td align="right">4912</td>
<td align="right">3155</td>
<td align="right">1.556894</td>
</tr>
<tr class="even">
<td align="left">B.1.1</td>
<td align="right">67683</td>
<td align="right">3770</td>
<td align="right">2603</td>
<td align="right">1.448329</td>
</tr>
<tr class="odd">
<td align="left">B.1.1.28</td>
<td align="right">8242</td>
<td align="right">829</td>
<td align="right">655</td>
<td align="right">1.265649</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="counts-of-mutations-in-100-bp-windows" class="section level1">
<h1>Counts of Mutations in 100 bp windows</h1>
<p>We do counts for the distinct time periods</p>
<p>Phase 1: until end of March 2020<br />
Phase 2: until 2020-08-31 Phase 3: after 2020-08-31</p>
<div id="for-phase-1" class="section level2">
<h2>for phase 1</h2>
<pre class="r"><code>data100 &lt;- mutations_annotated %&gt;%
  filter(species==&quot;Human&quot;) %&gt;%
  filter(date &lt; &quot;2020-04-01&quot;) %&gt;%
  filter(pangolin_lineage %in% to_extract)  %&gt;% # update
  group_by(pangolin_lineage, alt_gene_names, gene_bin, window_start, position, variant_base, type, aa_change) %&gt;%
  summarise(genomes_mutated=n()) %&gt;% # Number of genomes mutated pr single mutation
  filter(genomes_mutated &gt;= 1) %&gt;% # KEEP singletons 
  group_by(pangolin_lineage, alt_gene_names, gene_bin, type) %&gt;% 
  summarise(unique_mutations = n(), 
            max_genomes_mutated=max(genomes_mutated),
            window_start = mean(window_start)) # Count number of sites mutated pr window</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;alt_gene_names&#39;, &#39;gene_bin&#39;, &#39;window_start&#39;, &#39;position&#39;, &#39;variant_base&#39;, &#39;type&#39; (override with `.groups` argument)</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;alt_gene_names&#39;, &#39;gene_bin&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code>data100 &lt;- data100 %&gt;%  
  mutate(type=factor(type)) %&gt;%
  mutate(type = fct_recode(type,&quot;Synonymous substitutions&quot;=&quot;S&quot;, &quot;Nonsynonymous substitutions&quot;=&quot;N&quot;)) %&gt;%
  mutate(type = fct_relevel(type, &quot;Synonymous substitutions&quot;))</code></pre>
<p>Check that we have the lineages we wanted to extract</p>
<pre class="r"><code>dim(data100)</code></pre>
<pre><code>[1] 599   7</code></pre>
<pre class="r"><code>unique(data100$pangolin_lineage) %in% to_extract</code></pre>
<pre><code>[1] TRUE TRUE TRUE TRUE</code></pre>
<div id="overview-of-the-counts-for-extracted-lineages" class="section level3">
<h3>Overview of the counts for extracted lineages</h3>
<pre class="r"><code>ggplot(data100, aes(x=window_start, y=unique_mutations, size=max_genomes_mutated, color=type)) + 
  geom_point() +
  ylim(c(0,NA)) +
  xlim(c(0,NA)) +
  xlab(&quot;Genome position&quot;) +
  ylab(&quot;Number of unique mutations in 100bp&quot;) +
  scale_color_aaas(name=&quot;&quot;) +
  scale_size(&quot;Max genomes mutated&quot;, range=c(1,5)) +
  facet_wrap(~ pangolin_lineage, ncol = 1, scales = &quot;free_y&quot;)+
  theme(legend.position = &quot;none&quot;)+
  theme(legend.position = c(0.5,1),legend.justification = c(0.5,1))+
  # theme(legend.box.background = element_rect(fill=&quot;#F0F0F0&quot;)) +
  NULL</code></pre>
<p><img src="figure/S0_creating_data_bywindows_lineages_byphase.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="save-mutations-as-rds-object" class="section level3">
<h3>save mutations as RDS object</h3>
<pre class="r"><code>saveRDS(data100, &quot;output/data_100pb_pangolin_period1.rds&quot;)</code></pre>
</div>
</div>
<div id="for-period-2" class="section level2">
<h2>for period 2</h2>
<pre class="r"><code>data100 &lt;- mutations_annotated %&gt;%
  filter(species==&quot;Human&quot;) %&gt;%
  filter(date &gt;= &quot;2020-04-01&quot;) %&gt;%
  filter(date &lt; &quot;2020-08-31&quot;) %&gt;%
  filter(pangolin_lineage %in% to_extract)  %&gt;% # update
  group_by(pangolin_lineage, gene, gene_bin, window_start, position, variant_base,type, aa_change) %&gt;%
  summarise(genomes_mutated=n()) %&gt;% # Number of genomes mutated pr single mutation
  filter(genomes_mutated &gt; 1) %&gt;% # IGNORE singletons 
  group_by(pangolin_lineage, gene, gene_bin, type) %&gt;% 
  summarise(unique_mutations = n(), 
            max_genomes_mutated=max(genomes_mutated),
            window_start = mean(window_start)) # Count number of sites mutated pr window</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;gene&#39;, &#39;gene_bin&#39;, &#39;window_start&#39;, &#39;position&#39;, &#39;variant_base&#39;, &#39;type&#39; (override with `.groups` argument)</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;gene&#39;, &#39;gene_bin&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code>data100 &lt;- data100 %&gt;%  
  mutate(type=factor(type)) %&gt;%
  mutate(type = fct_recode(type,&quot;Synonymous substitutions&quot;=&quot;S&quot;, &quot;Nonsynonymous substitutions&quot;=&quot;N&quot;)) %&gt;%
  mutate(type = fct_relevel(type, &quot;Synonymous substitutions&quot;))

ggplot(data100, aes(x=window_start, y=unique_mutations, size=max_genomes_mutated, color=type)) + 
  geom_point() +
  ylim(c(0,NA)) +
  xlim(c(0,NA)) +
  xlab(&quot;Genome position&quot;) +
  ylab(&quot;Number of unique mutations in 100bp&quot;) +
  scale_color_aaas(name=&quot;&quot;) +
  scale_size(&quot;Max genomes mutated&quot;, range=c(1,5)) +
  facet_wrap(~ pangolin_lineage, ncol = 1, scales = &quot;free_y&quot;)+
  # theme(legend.position = c(0.5,1),legend.justification = c(0.5,1))+
  # theme(legend.box.background = element_rect(fill=&quot;#F0F0F0&quot;)) +
  NULL</code></pre>
<p><img src="figure/S0_creating_data_bywindows_lineages_byphase.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>unique(data100$pangolin_lineage)</code></pre>
<pre><code>[1] &quot;B.1&quot;      &quot;B.1.1&quot;    &quot;B.1.1.28&quot; &quot;B.1.1.7&quot;  &quot;B.1.177&quot; </code></pre>
<pre class="r"><code>table(data100$pangolin_lineage)</code></pre>
<pre><code>
     B.1    B.1.1 B.1.1.28  B.1.1.7  B.1.177 
     558      482      178       21       91 </code></pre>
<div id="save-mutations-as-rds-object-1" class="section level3">
<h3>save mutations as RDS object</h3>
<pre class="r"><code>saveRDS(data100, &quot;output/data_100pb_pangolin_period2.rds&quot;)</code></pre>
</div>
</div>
<div id="for-period-3" class="section level2">
<h2>for period 3</h2>
<pre class="r"><code>data100 &lt;- mutations_annotated %&gt;%
  filter(species==&quot;Human&quot;) %&gt;%
  filter(date &gt;= &quot;2020-09-01&quot;) %&gt;%
  filter(date &lt; &quot;2021-03-31&quot;) %&gt;%
  filter(pangolin_lineage %in% to_extract)  %&gt;% # update
  group_by(pangolin_lineage, gene, gene_bin, window_start, position, variant_base,type, aa_change) %&gt;%
  summarise(genomes_mutated=n()) %&gt;% # Number of genomes mutated pr single mutation
  filter(genomes_mutated &gt; 1) %&gt;% # IGNORE singletons 
  group_by(pangolin_lineage, gene, gene_bin, type) %&gt;% 
  summarise(unique_mutations = n(), 
            max_genomes_mutated=max(genomes_mutated),
            window_start = mean(window_start)) # Count number of sites mutated pr window</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;gene&#39;, &#39;gene_bin&#39;, &#39;window_start&#39;, &#39;position&#39;, &#39;variant_base&#39;, &#39;type&#39; (override with `.groups` argument)</code></pre>
<pre><code>`summarise()` regrouping output by &#39;pangolin_lineage&#39;, &#39;gene&#39;, &#39;gene_bin&#39; (override with `.groups` argument)</code></pre>
<pre class="r"><code>data100 &lt;- data100 %&gt;%  
  mutate(type=factor(type)) %&gt;%
  mutate(type = fct_recode(type,&quot;Synonymous substitutions&quot;=&quot;S&quot;, &quot;Nonsynonymous substitutions&quot;=&quot;N&quot;)) %&gt;%
  mutate(type = fct_relevel(type, &quot;Synonymous substitutions&quot;))

ggplot(data100, aes(x=window_start, y=unique_mutations, size=max_genomes_mutated, color=type)) + 
  geom_point() +
  ylim(c(0,NA)) +
  xlim(c(0,NA)) +
  xlab(&quot;Genome position&quot;) +
  ylab(&quot;Number of unique mutations in 100bp&quot;) +
  scale_color_aaas(name=&quot;&quot;) +
  scale_size(&quot;Max genomes mutated&quot;, range=c(1,5)) +
  facet_wrap(~ pangolin_lineage, ncol = 1, scales = &quot;free_y&quot;)+
  # theme(legend.position = c(0.5,1),legend.justification = c(0.5,1))+
  # theme(legend.box.background = element_rect(fill=&quot;#F0F0F0&quot;)) +
  NULL</code></pre>
<p><img src="figure/S0_creating_data_bywindows_lineages_byphase.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>unique(data100$pangolin_lineage)</code></pre>
<pre><code>[1] &quot;B.1&quot;      &quot;B.1.1&quot;    &quot;B.1.1.28&quot; &quot;B.1.1.7&quot;  &quot;B.1.177&quot;  &quot;B.1.351&quot;  &quot;P.1&quot;     </code></pre>
<pre class="r"><code>table(data100$pangolin_lineage)</code></pre>
<pre><code>
     B.1    B.1.1 B.1.1.28  B.1.1.7  B.1.177  B.1.351      P.1 
     604      601      398      614      611      520      420 </code></pre>
<div id="save-mutations-as-rds-object-2" class="section level3">
<h3>save mutations as RDS object</h3>
<pre class="r"><code>saveRDS(data100, &quot;output/data_100pb_pangolin_period3.rds&quot;)</code></pre>
</div>
</div>
<div id="check-in-name-of-the-orfs" class="section level2">
<h2>Check in name of the ORFs</h2>
<pre class="r"><code>unique(mutations_annotated$gene)</code></pre>
<pre><code> [1] &quot;nsp2&quot;              &quot;nsp3&quot;              &quot;nsp4&quot;             
 [4] &quot;nsp9&quot;              &quot;RDRp&quot;              &quot;S&quot;                
 [7] &quot;orf8&quot;              &quot;N&quot;                 &quot;exonuclease&quot;      
[10] &quot;nsp6&quot;              &quot;orf10&quot;             &quot;endoRNAse&quot;        
[13] &quot;methyltransferase&quot; &quot;orf3a&quot;             &quot;M&quot;                
[16] &quot;orf7a&quot;             &quot;leader&quot;            &quot;nsp10&quot;            
[19] &quot;helicase&quot;          &quot;E&quot;                 &quot;nsp8&quot;             
[22] &quot;proteinase&quot;        &quot;orf6&quot;              &quot;nsp7&quot;             
[25] &quot;nsp11&quot;             &quot;orf7b&quot;            </code></pre>
<pre class="r"><code>unique(gene_positions_md$gene)</code></pre>
<pre><code> [1] &quot;E&quot;                 &quot;endoRNAse&quot;         &quot;exonuclease&quot;      
 [4] &quot;helicase&quot;          &quot;leader&quot;            &quot;M&quot;                
 [7] &quot;methyltransferase&quot; &quot;N&quot;                 &quot;nsp10&quot;            
[10] &quot;nsp11&quot;             &quot;nsp2&quot;              &quot;nsp3&quot;             
[13] &quot;nsp4&quot;              &quot;nsp6&quot;              &quot;nsp7&quot;             
[16] &quot;nsp8&quot;              &quot;nsp9&quot;              &quot;orf10&quot;            
[19] &quot;orf3a&quot;             &quot;orf6&quot;              &quot;orf7a&quot;            
[22] &quot;orf7b&quot;             &quot;orf8&quot;              &quot;proteinase&quot;       
[25] &quot;RDRp&quot;              &quot;S&quot;                </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.1   ggsci_2.9       lubridate_1.7.9 cowplot_1.1.0  
 [5] forcats_0.5.0   stringr_1.4.0   dplyr_1.0.2     purrr_0.3.4    
 [9] readr_1.3.1     tidyr_1.1.1     tibble_3.0.3    ggplot2_3.3.2  
[13] tidyverse_1.3.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5       assertthat_0.2.1 rprojroot_2.0.2  digest_0.6.25   
 [5] utf8_1.1.4       R6_2.4.1         cellranger_1.1.0 backports_1.1.9 
 [9] reprex_0.3.0     evaluate_0.14    httr_1.4.2       highr_0.8       
[13] pillar_1.4.6     rlang_0.4.7      readxl_1.3.1     rstudioapi_0.11 
[17] whisker_0.4      blob_1.2.1       rmarkdown_2.3    labeling_0.3    
[21] munsell_0.5.0    broom_0.7.0      compiler_4.0.2   httpuv_1.5.4    
[25] modelr_0.1.8     xfun_0.16        pkgconfig_2.0.3  htmltools_0.5.0 
[29] tidyselect_1.1.0 workflowr_1.6.2  fansi_0.4.1      crayon_1.3.4    
[33] dbplyr_1.4.4     withr_2.2.0      later_1.1.0.1    grid_4.0.2      
[37] jsonlite_1.7.1   gtable_0.3.0     lifecycle_0.2.0  DBI_1.1.0       
[41] git2r_0.27.1     magrittr_1.5     scales_1.1.1     cli_2.0.2       
[45] stringi_1.4.6    farver_2.0.3     fs_1.5.0         promises_1.1.1  
[49] xml2_1.3.2       ellipsis_0.3.1   generics_0.0.2   vctrs_0.3.2     
[53] tools_4.0.2      glue_1.4.1       hms_0.5.3        yaml_2.2.1      
[57] colorspace_1.4-1 rvest_0.3.6      knitr_1.29       haven_2.3.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
