---
title: "Changes in mutational patterns over time"
author: "Palle Villesen"
date: 'Last update: `r Sys.time()`'
output: 
  html_document:
    theme: united
    code_folding: hide
    toc: true
    toc_depth: 2 
editor_options: 
  chunk_output_type: console
--- 

```{r}
source("analysis/hard_coded_params.R")
# data_path <- "data/2021-06-18/" 
# use_downsampled <- TRUE

```

# Using data version: `r data_path`

# Init

```{r, message=F, warning=F}

library(tidyverse)
library(cowplot)  # For theme
library(lubridate)
library(ggsci)
library(ggrepel)


text_base_size   <- 8
fig.witdh        <- 210
fig.height       <- 160

ggplot_text_size <- text_base_size / ggplot2::.pt
# Now use: geom_text(..., size = ggplot_text_size)
theme_set(theme_cowplot(font_size = text_base_size, 
                        rel_small = 1, rel_tiny = 1, 
                        rel_large = 1))

knitr::opts_chunk$set(fig.width = fig.witdh/25.4)
knitr::opts_chunk$set(fig.height = fig.height/25.4)
knitr::opts_chunk$set(dpi = 108)

if (use_downsampled) {
  mutations <- read_rds(paste(data_path, "/mutations_snpeff_annotated_tidy_300K_downsampled.rds", sep=""))
  metadata  <- read_rds(paste(data_path, "/metadata_snpeff_tidy_300K_downsampled.rds",sep=""))
  } else {
  mutations <- read_rds(paste(data_path, "/mutations_snpeff_annotated_tidy.rds", sep=""))
  metadata  <- read_rds(paste(data_path, "/metadata_snpeff_tidy.rds",sep=""))
}

x <- Sys.setlocale(locale = "English")
rm(x)
```

# Mutations in specific countries

```{r}

# Get mutated genomes in spike
interesting_mutations <- c("D614G", "N501Y", "E484K", "E484Q", "T478K","L452R", "P681R") #
interesting_countries <- c("United Kingdom", "Denmark", "Portugal")

transitions   <- c("C>T", "T>C", "A>G", "G>A")
transversions <- c("C>A", "C>G", "T>A", "T>G", "A>C", "A>T", "G>T", "G>C")

interesting_lineages <- c("B.1",
                          "B.1.1.7",# British
                          "B.1.351", # South Africa
                          "B.1.617.2" # Indian variant
                          )

# Tag each genome with each of these mutations
tmp1 <- metadata %>%
  filter(species=="Human" & country %in% interesting_countries) %>%
  mutate(date_bin = cut(date, breaks = "7 days") %>% ymd())
  
# For each mutation, get proportion

results <- tibble()

for(i in 1:length(interesting_mutations)) {
  
  tmp2 <- mutations %>%
    filter(gene=="S" & aa_change==interesting_mutations[i])
  
  tmp3 <- tmp1 %>%
    mutate(mutation = interesting_mutations[i]) %>%
    mutate(has_it = ifelse(id %in% tmp2$id, "mutated", "not_mutated")) %>%
    group_by(country, date_bin, mutation) %>%
     summarise(total = n(),
               prop = sum(has_it=="mutated") / n())

  results <- results %>% bind_rows(tmp3)
}

pd <- results

ggplot(pd, aes(x=date_bin, y=prop,size=total, color=mutation)) + 
  geom_point() +
  geom_line(size=1) +
  xlab("") +
  ylab("Proportion of genomes with mutation") +
  scale_size("Number of genomes") +
  ggsci::scale_color_aaas(name="") +
  facet_wrap( ~country, ncol=2) +
  NULL

```

# Lineages in DK & UK

```{r}

tmp <- metadata %>% 
  filter(species=="Human" & country %in% interesting_countries) %>%
  mutate(pangolin2 = ifelse(pangolin_lineage %in% interesting_lineages, pangolin_lineage, "Other")) %>%
  mutate(date_bin = cut(date, breaks = "7 days") %>% ymd()) %>%
  group_by(date_bin, country) %>%
  count(pangolin2) %>%
  {.}

pd <- tmp %>% 
  group_by(date_bin, country) %>%
  summarise(total = sum(n)) %>%
  left_join(tmp) %>%
  mutate(prop = n/total)

ggplot(pd, aes(x=date_bin, y=prop, size=total, color=pangolin2)) + 
  geom_point() +
  geom_line(size=1) +
  xlab("") +
  ylab("Proportion of genomes with mutation") +
  scale_size("Number of genomes") +
  ggsci::scale_color_aaas(name="") +
  facet_wrap(~country, ncol=2) +
  NULL

```



# DK analysis: Mutational clock (Syn vs. Non-Syn)

```{r}


tmp <- metadata %>% 
  filter(species=="Human" & country %in% interesting_countries) %>%
  filter(pangolin_lineage %in% interesting_lineages) %>%
  mutate(date_bin = cut(date, breaks = "2 weeks") %>% ymd()) %>%
  {.}

pd <-  tmp %>%
  group_by(date_bin,country, pangolin_lineage, count_N, count_S) %>%
  count() %>%
  pivot_longer(names_to = "type", 
              values_to = "mutations", 
                   cols = c(count_N, count_S)) %>%
  mutate(type = fct_recode(type, 
                           "Synonymous mutations"="count_S",
                           "Nonsynonymous mutations"="count_N"
                           ))

ggplot(pd, aes(x=date_bin, y=mutations, color=pangolin_lineage, size=n)) +
  geom_point(alpha=0.5) + 
  geom_smooth(method="lm", se=F, size=1, show.legend = F) +
  ylab("Accumulated mutations") +
  facet_grid(country ~ type) +
  theme(axis.title.x = element_blank()) +
  scale_color_aaas(name="Lineage") +
  scale_size_continuous(name = "Genomes") +
  theme(strip.background = element_blank()) +
  ggtitle(paste(nrow(tmp), " genomes", sep="")) +
  NULL

```


## Lineage analysis 

Remove nearly fixed mutations


```{r}


tmp1 <- metadata %>% 
  filter(species=="Human" & country %in% interesting_countries) %>%
  filter(pangolin_lineage %in% interesting_lineages) %>%
  mutate(date_bin = cut(date, breaks = "2 weeks") %>% ymd())

# Genomes pr lineage
tmp2 <- tmp1 %>%
  group_by(pangolin_lineage) %>%
  summarise(genomes=n())

# count of genomes mutated pr mutation
tmp3 <- tmp1 %>% 
  select(id, pangolin_lineage) %>%
  left_join(mutations) %>% 
  select(pangolin_lineage, id, ref_base, variant_base, position) %>%
  distinct() %>%
  group_by(pangolin_lineage, ref_base, variant_base, position) %>%
  summarise(genomes_mutated = n()) %>%
  mutate(base_change = paste(ref_base, variant_base, sep=">")) %>%
  mutate(mutation_type = ifelse(base_change %in% transitions, "transition", NA)) %>%
  mutate(mutation_type = ifelse(base_change %in% transversions, "transversion", mutation_type))

# Good mutations
tmp4 <- tmp2 %>% 
  left_join(tmp3) %>% 
  filter(genomes_mutated/genomes < 0.9)

# Getting summary of each genome
tmp5 <- tmp4 %>% 
  left_join(mutations) %>% # This will limit to only the good mutations
  filter(gene!="ORF1ab_pp1a") %>%
  group_by(id) %>% # Summarise pr. genome
  summarise(count_ts = sum(base_change %in% transitions),
            count_tv = sum(base_change %in% transversions),
            count_non = sum(type=="N"),
            count_syn = sum(type=="S")
            ) %>%
  filter(count_syn > 5) %>%
  filter(count_tv > 5) %>%
  mutate(tstv_ratio = count_ts/count_tv) %>%
  mutate(ns_ratio = count_non/count_syn) %>%
  filter(tstv_ratio > 0) %>%
  filter(ns_ratio > 0) %>%
  {.}

pd <- tmp1 %>% left_join(tmp5)

ggplot(pd,aes(x=date_bin, y=tstv_ratio)) + 
  geom_jitter(width=1, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), 
               fun=median, colour="red", geom="line") +
  facet_grid(country ~ pangolin_lineage) +
  ylab("Nonsynonymous / synonymous changes") +
  theme(axis.title.x = element_blank())+
  NULL

ggplot(pd,aes(x=date_bin, y=ns_ratio)) + 
  geom_jitter(width=1, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), fun=median, colour="red", geom="line") +
  facet_grid(country ~ pangolin_lineage) +
  ylab("Transitions / Transversions") +
  theme(axis.title.x = element_blank()) +
  NULL

pd2 <- pd %>% pivot_longer(cols = c(count_ts, count_tv))

ggplot(pd2,aes(x=date_bin, y=value)) + 
  geom_jitter(width=0.5, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), fun=median, colour="red", geom="line") +
  facet_grid(name~pangolin_lineage) +
  theme(axis.title = element_blank()) +
  NULL

pd2 <- pd %>% pivot_longer(cols = c(count_syn, count_non))

ggplot(pd2,aes(x=date_bin, y=value)) + 
  geom_jitter(width=1, height = 0, alpha=0.5) +
  stat_summary(mapping = aes(group=pangolin_lineage), fun=median, colour="red", geom="line") +
  facet_grid(name~pangolin_lineage) +
  theme(axis.title = element_blank()) +
  NULL


```


